moveit_core/CMakeLists.txt

```diff
@@ -4,9 +4,14 @@ project(moveit_core)
 add_compile_options(-std=c++11)
 
 # Warnings
-add_definitions(-Wall -Wextra
+add_compile_options(-Wall -Wextra
   -Wwrite-strings -Wunreachable-code -Wpointer-arith -Wredundant-decls -Wcast-qual
-  -Wno-unused-parameter -Wno-unused-function -Wno-maybe-uninitialized)
+  -Wno-unused-parameter -Wno-unused-function)
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  add_compile_options(-Wno-maybe-uninitialized)
+elseif(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+  add_compile_options(-Wno-inconsistent-missing-override -Wno-potentially-evaluated-expression)
+endif()
```

Comment:
It would be good to have comments in here on why these `-Wno-*` flags were added.
Are these warnings something we should improve in our code base?

Reply:
These flags improve our code style by throwing warnings for common issues. See my PR https://github.com/ros-planning/moveit/pull/971

I'd like to add this to all the packages in moveit

Reply:
No they don't `-Wno-*` actually *disables* specific warnings. Probably because @rhaschke either decided they are not important for us, or because some package we rely on throws lots of warnings we can't fix ourselves.